x/programs/examples/token.go

```diff
@@ -58,23 +57,23 @@ func (t *Token) Run(ctx context.Context) error {
 	)
 
 	// simulate create program transaction
-	programID := ids.GenerateTestID()
-	err = storage.SetProgram(ctx, t.db, programID, t.programBytes)
+	programAddress := codec.CreateAddress(runtime.ED25519ID, ids.GenerateTestID())
+	err = storage.SetProgram(ctx, t.db, programAddress, t.programBytes)
 	if err != nil {
 		return err
 	}
 
-	programIDPtr, err := runtime.WriteBytes(rt.Memory(), programID[:])
+	programAddressPtr, err := argumentToSmartPtr(programAddress, rt.Memory())
 	if err != nil {
 		return err
 	}
 
 	t.log.Debug("new token program created",
-		zap.String("id", programID.String()),
+		zap.String("id", codec.MustAddressBech32("token_", programAddress)),
```

Comment:
usually, there is no trailing dash here...there is already a "separator" character in bech32.